### PR TITLE
include doc on how to bring your own certificate with net-contour

### PR DESCRIPTION
**Pull request by dprotaso**
Documents this feature https://github.com/knative-sandbox/net-contour/pull/193

### CODE REVIEW

**Comment by dprotaso** [comment]
/hold
will wait for the PR to land

File: docs/serving/using-a-tls-cert.md [hunk 1]
```diff
@@ -153,6 +153,56 @@ For instructions about enabling Knative for automatic certificate provisioning,
 see [Enabling automatic TLS cert provisioning](./using-auto-tls.md). Otherwise,
 continue below for instructions about manually adding a certificate.
 
+{{< tabs name="serving_networking" default="Istio" >}}
+{{% tab name="Contour" %}}
+
+To manually add a TLS certificate to your Knative cluster, you create a
+Kubernetes secret and then configure the `net-contour`
```
Comment (mattmoor) [inline docs/serving/using-a-tls-cert.md:160]: The last bit reads like it's missing a word.

File: docs/serving/using-a-tls-cert.md [hunk 2]
```diff
@@ -153,6 +153,56 @@ For instructions about enabling Knative for automatic certificate provisioning,
...
+1. Run the following command to create a Kubernetes secret to hold your TLS
+   certificate, `cert.pem`, and the private key, `cert.pk`:
+
+   ```shell
+   kubectl create --namespace contour-external tls default-cert \
+     --key cert.pem \
+     --cert cert.pem
+   ```
+
+   where `cert.pem` and `cert.pem` are your certificate and private key files.
```
Comment (mattmoor) [inline docs/serving/using-a-tls-cert.md:171]: did you mean to name these the same?

File: docs/serving/using-a-tls-cert.md [hunk 3]
```diff
@@ -153,6 +153,56 @@ For instructions about enabling Knative for automatic certificate provisioning,
...
+1. Update net-contour settings to leverage the secret 
+   
+   ```shell
+   kubectl edit cm config-contour -n knative-serving
+   ```
+
+   add the following data property
+
+   ```yaml
+   default-tls-secret-name: "contour-external/default-cert"
```
Comment (mattmoor) [inline docs/serving/using-a-tls-cert.md:200]: Provide a PATCH command?

File: docs/serving/using-a-tls-cert.md [hunk 4]
```diff
@@ -153,6 +153,56 @@ For instructions about enabling Knative for automatic certificate provisioning,
...
+   kind: TLSCertificateDelegation
+   metadata:
+     name: default-delegation
+     namespace: contour-external
+   spec:
+     delegations:
+       - secretName: default-cert
+         targetNamespaces:
+         - "*"
+   ```
```
Comment (mattmoor) [inline docs/serving/using-a-tls-cert.md:189]: Do you always need this, or just when Contour is in some mode?

File: docs/serving/using-a-tls-cert.md (hunk 1 above)
Comment (abrennan89) [inline docs/serving/using-a-tls-cert.md:160]:
```suggestion
To manually add a TLS certificate to your Knative cluster, you must create a
Kubernetes secret and then configure `net-contour`.
```

Comment (abrennan89) [reply docs/serving/using-a-tls-cert.md:160]: Can we maybe explain what `net-contour` is here? Is this a literal setting?

File: docs/serving/using-a-tls-cert.md [hunk 5]
```diff
@@ -153,6 +153,56 @@ For instructions about enabling Knative for automatic certificate provisioning,
...
 continue below for instructions about manually adding a certificate.
 
+{{< tabs name="serving_networking" default="Istio" >}}
+{{% tab name="Contour" %}}
+
+To manually add a TLS certificate to your Knative cluster, you create a
+Kubernetes secret and then configure the `net-contour`
+
+1. Run the following command to create a Kubernetes secret to hold your TLS
+   certificate, `cert.pem`, and the private key, `cert.pk`:
```
Comment (abrennan89) [inline docs/serving/using-a-tls-cert.md:163]:
```suggestion
1. Create a Kubernetes secret to hold your TLS certificate, `cert.pem`, and the private key, `cert.pk`, by entering the following command:
```

File: docs/serving/using-a-tls-cert.md [hunk 6]
```diff
@@ -153,6 +153,56 @@ For instructions about enabling Knative for automatic certificate provisioning,
...
+   ```shell
+   kubectl create --namespace contour-external tls default-cert \
+     --key cert.pem \
+     --cert cert.pem
+   ```
+
+   where `cert.pem` and `cert.pem` are your certificate and private key files.
+
+   You'll need to keep note of the namespace and secret name for the next two 
+   steps.
```
Comment (abrennan89) [inline docs/serving/using-a-tls-cert.md:174]:
```suggestion
  **IMPORTANT:** Take note of the namespace and secret name. You will need these in future steps.
```

File: docs/serving/using-a-tls-cert.md [hunk 7]
```diff
@@ -153,6 +153,56 @@ For instructions about enabling Knative for automatic certificate provisioning,
...
+
+1. Create a delegation to allow the secret to be used by your developers
```
Comment (abrennan89) [inline docs/serving/using-a-tls-cert.md:176]:
```suggestion
1. Create a delegation to allow the secret to be used:
```

File: docs/serving/using-a-tls-cert.md [hunk 8]
```diff
@@ -153,6 +153,56 @@ For instructions about enabling Knative for automatic certificate provisioning,
...
+
+To manually add a TLS certificate to your Knative cluster, you create a
+Kubernetes secret and then configure the `net-contour`
+
+1. Run the following command to create a Kubernetes secret to hold your TLS
+   certificate, `cert.pem`, and the private key, `cert.pk`:
+
+   ```shell
+   kubectl create --namespace contour-external tls default-cert \
+     --key cert.pem \
```
Comment (abrennan89) [inline docs/serving/using-a-tls-cert.md:167]:
```suggestion
     --key cert.pk \
```

File: docs/serving/using-a-tls-cert.md (hunk 4 above)
Comment (abrennan89) [reply docs/serving/using-a-tls-cert.md:189]: It's not really clear in each step why a user or admin needs to do each of these things IMHO

File: docs/serving/using-a-tls-cert.md [hunk 9]
```diff
@@ -153,6 +153,56 @@ For instructions about enabling Knative for automatic certificate provisioning,
...
+     name: default-delegation
+     namespace: contour-external
+   spec:
+     delegations:
+       - secretName: default-cert
+         targetNamespaces:
+         - "*"
+   ```
+
+1. Update net-contour settings to leverage the secret
```
Comment (abrennan89) [inline docs/serving/using-a-tls-cert.md:191]: Again, what is net-contour? Can we use consistent formatting to make it obvious if this is a literal setting or a concept? Earlier in the doc it's written as `net-contour` which looks like a literal setting, but then here it isn't?

Please use punctuation at the end of each numbered list entry also. e.g.

1. Create x thing.
2. Create x thing by entering the command:

File: docs/serving/using-a-tls-cert.md [hunk 10]
```diff
@@ -153,6 +153,56 @@ For instructions about enabling Knative for automatic certificate provisioning,
...
+         - "*"
+   ```
+
+1. Update net-contour settings to leverage the secret 
+   
+   ```shell
+   kubectl edit cm config-contour -n knative-serving
+   ```
+
+   add the following data property
```
Comment (abrennan89) [inline docs/serving/using-a-tls-cert.md:197]: Should be its own step.

**Comment by dprotaso** [comment]
PTAL again.

I've clarified that `net-contour` is the `Knative Contour plugin`

File: docs/serving/using-a-tls-cert.md (hunk 4 above)
Comment (dprotaso) [reply docs/serving/using-a-tls-cert.md]: i've added clarifications

**Comment by dprotaso** [comment]
link failure = xip.io

**Comment by abrennan89** [comment]
/lgtm
/approve

Thanks @dprotaso , looks good!
I wouldn't worry about that particular link check failure, there are a few things causing known issues for us at the moment and we're going to revisit automation and things like these tests for docs when we can. We don't need to block this PR on it :wink:

**Comment by dprotaso** [comment]
/retest

integration tests should be fixed

**Pull request merged**